### PR TITLE
feat(admin) - wire in queries and rendering results

### PR DESCRIPTION
**Pull request by onewland**
Construct queries and send them to the API, and display the query result history in a table

### CODE REVIEW

File: snuba/admin/static/clickhouse_queries/types.tsx [hunk 1]
```diff
@@ -9,4 +9,27 @@ type ClickhouseNodeData = {
   local_nodes: ClickhouseNode[];
 };
 
-export { ClickhouseNodeData };
+type ClickhouseCannedQuery = {
+  description?: string;
+  name: string;
+  sql?: string;
```
Comment (lynnagara) [inline snuba/admin/static/clickhouse_queries/types.tsx:15]: How come description and sql are optional, shouldn't we always get this from the API?

File: snuba/admin/static/api_client.tsx [hunk 2]
```diff
@@ -57,6 +64,25 @@ function Client() {
...
+          method: "POST",
+          body: new Blob([JSON.stringify(query)], { type: "application/json" }),
+        })
+      )
+        .then((resp) => resp.json())
+        .then((resp: QueryResult) => {
+          // used for keying in history
+          resp.timestamp = Date.now();
+          return resp;
+        });
```
Comment (lynnagara) [inline snuba/admin/static/api_client.tsx:84]: Can you remove this? Seems like it's not being used

File: snuba/admin/static/api_client.tsx [hunk 3]
```diff
@@ -57,6 +64,25 @@ function Client() {
...
+    getClickhouseCannedQueries: () => {
+      const url = baseUrl + "clickhouse_queries";
+      return fetch(url).then((resp) => resp.json());
+    },
+    executeQuery: (query: QueryRequest) => {
+      const url = baseUrl + "run_clickhouse_system_query";
+      return fetch(
+        new Request(url, {
+          method: "POST",
+          body: new Blob([JSON.stringify(query)], { type: "application/json" }),
```
Comment (lynnagara) [inline snuba/admin/static/api_client.tsx:76]: Why blob? Why not just post the JSON?

File: snuba/admin/static/clickhouse_queries/index.tsx [hunk 4]
```diff
@@ -44,9 +51,27 @@ function ClickhouseQueries(props: { api: Client }) {
...
+        query_name: queryName,
+      };
+    });
+  }
+
+  function executeQuery() {
+    props.api.executeQuery(query as QueryRequest).then((result) => {
+      result.input_query = `${query.query_name}(${query.storage},${query.host}:${query.port})`;
+      setQueryResultHistory((prevHistory) => [result, ...prevHistory]);
+      console.log(queryResultHistory);
```
Comment (lynnagara) [inline snuba/admin/static/clickhouse_queries/index.tsx:67]: nit: Remove this

File: snuba/admin/static/table.tsx [hunk 5]
```diff
@@ -5,13 +5,15 @@ import { COLORS } from "./theme";
 type TableProps = {
   headerData: ReactNode[];
   rowData: ReactNode[][];
-  columnWidths: number[];
+  columnWidths?: number[];
 };
 
 function Table(props: TableProps) {
-  const { headerData, rowData, columnWidths } = props;
+  let { headerData, rowData, columnWidths } = props;
```
Comment (lynnagara) [inline snuba/admin/static/table.tsx:12]: nit: Why changing this? I normally prefer const unless reassigning later

File: snuba/admin/static/table.tsx [hunk 6]
```diff
@@ -5,13 +5,15 @@ import { COLORS } from "./theme";
...
 
-  const sumColumnWidths = columnWidths.reduce((acc, i) => acc + i, 0);
+  const autoColumnWidths = Array(headerData.length).fill(1);
+  const notEmptyColumnWidths = columnWidths ?? autoColumnWidths;
```
Comment (lynnagara) [inline snuba/admin/static/table.tsx:15]: TIL about ??

**Review by lynnagara — COMMENTED** [review]
For some reason this does not work for me. I get 500 errors and `TypeError: cannot unpack non-iterable ClickhouseResult object` error when making requests via this UI.

File: snuba/admin/static/api_client.tsx (hunk 3 above)
Comment (onewland) [reply snuba/admin/static/api_client.tsx:76]: I just copied this example from MDN: https://developer.mozilla.org/en-US/docs/Web/API/Request/blob#examples

File: snuba/admin/static/table.tsx (hunk 5 above)
Comment (onewland) [reply snuba/admin/static/table.tsx:12]: in my first version, i reassigned columnWidths. will update

File: snuba/admin/static/clickhouse_queries/types.tsx (hunk 1 above)
Comment (onewland) [reply snuba/admin/static/clickhouse_queries/types.tsx:15]: description is not present for one of the predefined queries.

sql should always be, not sure why I did that

**Comment by onewland** [comment]
> For some reason this does not work for me. I get 500 errors and `TypeError: cannot unpack non-iterable ClickhouseResult object` error when making requests via this UI.

This seemed to be caused by a merge from master, I'll look into it

File: snuba/admin/static/api_client.tsx (hunk 3 above)
Comment (lynnagara) [reply snuba/admin/static/api_client.tsx:76]: I think we should just send JSON like the rest of the API
```suggestion
return fetch(
  headers: { "Content-Type": "application/json" },
  url,
  method: "POST",
  body: JSON.stringify(query)
)
```

File: snuba/admin/static/clickhouse_queries/types.tsx (hunk 1 above)
Comment (lynnagara) [reply snuba/admin/static/clickhouse_queries/types.tsx:15]: It seems like the description key is actually always there but it is nullable: https://github.com/getsentry/snuba/blob/master/snuba/admin/clickhouse/system_queries.py#L54.
So the type should maybe be `string | null`

**Review by lynnagara — APPROVED** [review]

**Pull request merged**